Add Recommended Build Ideas

ChangoMan opened the pull request:
- Moves 2 challenges, the Multisig and SVG NFT challenges, out of the "Advanced" challenges section.
- Create 2 build ideas that will show up if a user doesn't have any builds. Only shows on your own profile.
- When viewing another person's profile with no builds, nothing shows up.
- If there are builds, show builds normally.
- Tweek the BuildCard section so it takes up the full amount of space in the grid.

<img width="1919" height="960" alt="Screenshot 2025-07-22 at 7 23 51 PM" src="https://github.com/user-attachments/assets/f1142dfe-68e1-4211-a8c3-7dfbd91da8c1" />

carletex reviewed (commented):
This looks great, Hunter!

Pretty inlined with https://github.com/BuidlGuidl/SpeedRunEthereum-v2/issues/249 (even if you started before the issue was created)

A few things:

- Let's have a more warm/encouraging message as the intro text

I suck at this but something like:

"Showcase your Ethereum builds and share them with the community!
If you’re just getting started or looking for your next idea, take a look at some recommended builds below."

(maybe we can ask GPT for help)

- Regarding the cards (vs plain links). Open to both, but cards looks cool! Would it make sense to have some kind of opacity for them? (and remove it when hover?)
e.g.

<img width="443" height="463" alt="image" src="https://github.com/user-attachments/assets/57cd7c03-3740-42d5-a46d-3ac5a3dbd12b" />
(just to make it more apparent)

- From #249: 
 
> Don't let users with no challenges completed to submit Builds. We can show the "New Build" button disabled with a tooltip encouraging to complete their first challenge to submit a build."

Could we add it? Front-end + back-end validation.

-  (Discussion) When the user adds a build, the ideas disappear. Wondering if we should still have them somewhere (even if there's a collapsed accorddion that shows just the title + link...and only visible for the user connected in their profile). Just thinking that if we end adding 5+ ideas, it'd be nice to have access to them.

cc @Pabl0cks

ChangoMan commented on packages/nextjs/public/assets/builds/build-multisig.png:
Oh yes, I can use the existing SVGs! They'll probably look sharper than the image

ChangoMan commented:
@carletex Thanks for the feedback, added some updates:

- Warmer message (original message was from ai suggestion). I like the one you gave
- Use existing challenge svgs in the build ideas card. Delete the png images
- Added opacity and hover effects to card
- Added blocking for submitting a build, if you do not have a challenge completed.

I feel like we can leave the challenges on the homepage for now, it doesn't really hurt? Plus I hear Austin shill the multi-sig challenge a lot during his talks. We can remove later too, not sure if it needs a bigger conversation about the homepage.

carletex commented:
Thanks for the updates, Hunter!!

Pushed a couple tweaks:
- Check if user has COMPLETED a challenge (vs just submitted, which could be "rejected"
- We were using `isOwner`, `isProfileOwner` and `isUserProfile` for the same thing in different places. I kept `isProfileOwner`
- Added some extra margin. If not, I feel that the "advanced section" is super easy to miss and close to "builds". Related to https://github.com/BuidlGuidl/SpeedRunEthereum-v2/issues/244 (it might fix this)

>  I feel like we can leave the challenges on the homepage for now, it doesn't really hurt? Plus I hear Austin shill the multi-sig challenge a lot during his talks. We can remove later too, not sure if it needs a bigger conversation about the homepage.

Fair! Let's have that discussion on #237 

Related: 

> (Discussion) When the user adds a build, the ideas disappear. Wondering if we should still have them somewhere (even if there's a collapsed accorddion that shows just the title + link...and only visible for the user connected in their profile). Just thinking that if we end adding 5+ ideas, it'd be nice to have access to them.

Once you submit a build, you lose access to the idea list. Let's open a new issue for it.

carletex reviewed (approved):
Merging, good stuff!
